src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -712,45 +720,43 @@ private async Task Orphan(IHost instance, CancellationToken cancellationToken =
+                _logger.LogError(ex, "Failed to stop host instance '{hostInstanceId}'.", hostInstanceId);
             }
             finally
             {
-                if (instance != null)
-                {
-                    GetHostLogger(instance).LogDebug("Disposing ScriptHost.");
+                GetHostLogger(instance).LogDebug("Disposing ScriptHost: '{hostInstanceId}'", hostInstanceId);
 
-                    if (isStandbyHost && !_scriptWebHostEnvironment.InStandbyMode)
+                if (isStandbyHost && !_scriptWebHostEnvironment.InStandbyMode)
+                {
+                    // For cold start reasons delay disposing script host if specializing out of placeholder mode
+                    Utility.ExecuteAfterColdStartDelay(_environment, () =>
                     {
-                        // For cold start reasons delay disposing script host if specializing out of placeholder mode
-                        Utility.ExecuteAfterColdStartDelay(_environment, () =>
+                        try
                         {
-                            try
-                            {
-                                GetHostLogger(instance).LogDebug("Starting Standby ScriptHost dispose");
-                                instance.Dispose();
-                                _logger.LogDebug("Standby ScriptHost disposed");
-                            }
-                            catch (Exception e)
-                            {
-                                _logger.LogError(e, "Failed to dispose Standby ScriptHost instance");
-                                throw;
-                            }
-                        }, cancellationToken);
-
-                        _logger.LogDebug("Standby ScriptHost marked for disposal");
-                    }
-                    else
-                    {
-                        DisposeDependencyTrackingModule(instance);
-                        instance.Dispose();
-                        _logger.LogDebug("ScriptHost disposed");
-                    }
+                            GetHostLogger(instance).LogDebug("Starting Standby ScriptHost dispose");
+                            instance.Dispose();
+                            _logger.LogDebug("Standby ScriptHost disposed");
+                        }
+                        catch (Exception e)
+                        {
+                            _logger.LogError(e, "Failed to dispose Standby ScriptHost instance");
+                            throw;
+                        }
+                    }, cancellationToken);
+
+                    _logger.LogDebug("Standby ScriptHost marked for disposal");
+                }
+                else
+                {
+                    DisposeDependencyTrackingModule(instance);
+                    instance.Dispose();
+                    _logger.LogDebug("ScriptHost disposed: '{hostInstanceId}'", hostInstanceId);
```

Comment:
This is one of those cases where I do feel we have a number of queries and potentially detectors relying on this log message, and this change could cause them to break. Thoughts?

Reply:
I was trying to keep it so that those could run with "startswith" -- but I do agree. Let me see if there's another way I can add this information.